lessons/configuration-dotnet.md

```diff
@@ -104,19 +102,148 @@ Include using GitHub secret in GH action in the Deployment lesson.
+
+### Steps - Setting up our function to use the App Configuration
+
+Now that we have our App Configuration set up with a configuration value, lets use it on our application!
+
+We have some changes that we need to make to our code in order to make this work
+
+#### Steps - Setup dependency injection
+
+In order to use the App Configuration in our code we first need to enable dependency injection for the function, and set up out dependency injection container for the App Configuration
+
+1. Add the following NuGet packages
+
+* Microsoft.Extensions.Configuration.AzureAppConfiguration
+* Microsoft.Azure.Functions.Extensions
+
+2. Add a new class `FunctionStartup.cs` to the route of the project with the following `using` statements
+```c#
+using System;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+```
+3. Add the Assembly attribute to register the start-up method for the function app, and allow us set-up the dependency injection
+```c#
+[assembly: FunctionsStartup(typeof(AzureFunctionsUniversity.Demo.Configuration.Startup))]
+```
+
+3. Add namespace and class
+
+```c#
+namespace AzureFunctionsUniversity.Demo.Configuration
+{
+	class Startup : FunctionsStartup
+	{
+
+	}
+}
+```
+
+4. Inside the class add the `ConfigureAppConfiguration` function<br />This connects our application to our App Configuration using an environment variable to hold the connectin string we just copied.
+```c#
+		public override void ConfigureAppConfiguration(IFunctionsConfigurationBuilder builder)
+		{
+			builder.ConfigurationBuilder.AddAzureAppConfiguration(options =>
+			{
+				options.Connect(Environment.GetEnvironmentVariable("AppConfigurationConnectionString"));
+			});
+
+		}
+```
+
+5. Add the `Configure` function to add the Azure App Configuration to the dependency injection container for use in our function
+```c#
+		public override void Configure(IFunctionsHostBuilder builder)
+		{
+			builder.Services.AddAzureAppConfiguration();
+		}
+```
+
+6. Finally, we need to set the local setting for the connection string. Add the following setting to the `local.settings.json` file in the `values` section
+```json
+"AppConfigurationConnectionString": "Endpoint=https://azure-university-app-config.azconfig.io;..."
+```
+
+#### Steps - Create the a second function to use the App Configuration
+
+***TODO***
+
+1. Add a new http troggered function to the application
+2. Remove the static keyword from the Azure Function class/method - as we are using dependency injection we need a constructor for the class
+3. Create a constructor for the class, pass in an IConfiguration and set a private field to hold the value
+4. Replace the function method with the code below to return the value from the Azure Configuration
+```c#
+[FunctionName("ReadingAppConfigurationVariables")]
```

Comment:
Shall we use `nameof(<Function Classname>)` inside the `FunctionName` attribute instead of the string? That way it's more inline with the other lessons.

Reply:
Ah, I missed that, sorry! I'll make the change to the source code and md file!

As for the dynamic config, that sounds great! I was going to put it in this lesson, but decided to keep it simple. Building on it further sounds like a great idea! I fyou can prepare it I'd appriciate it - I'm fighting writers block for the local settings right now 😅